src/platform/FailSafeContext.cpp

```diff
@@ -37,36 +37,32 @@ constexpr TLV::Tag kUpdateNocCommandTag = TLV::ContextTag(2);
 
 void FailSafeContext::HandleArmFailSafeTimer(System::Layer * layer, void * aAppState)
 {
-    FailSafeContext * context = reinterpret_cast<FailSafeContext *>(aAppState);
-    context->FailSafeTimerExpired();
+    FailSafeContext * failSafeContext = reinterpret_cast<FailSafeContext *>(aAppState);
+    failSafeContext->FailSafeTimerExpired();
 }
 
 void FailSafeContext::HandleDisarmFailSafe(intptr_t arg)
 {
-    FailSafeContext * this_ = reinterpret_cast<FailSafeContext *>(arg);
-
-    this_->mFailSafeBusy = false;
-
-    if (ConfigurationMgr().SetFailSafeArmed(false) != CHIP_NO_ERROR)
-    {
-        ChipLogError(DeviceLayer, "Failed to set FailSafeArmed config to false");
-    }
-
-    if (DeleteFromStorage() != CHIP_NO_ERROR)
-    {
-        ChipLogError(DeviceLayer, "Failed to delete FailSafeContext from configuration");
-    }
+    FailSafeContext * failSafeContext = reinterpret_cast<FailSafeContext *>(arg);
+    failSafeContext->DisarmFailSafe();
 }
 
 void FailSafeContext::FailSafeTimerExpired()
 {
+    if (!IsFailSafeArmed())
+    {
+        // In case this was a pending timer event in event loop, and we had
```

Comment:
`FailSafeContext::DisarmFailSafe` cancels the timer. I guess the point is that if we're in the middle of timer processing the CancelTimer might actually not do anything because the timer has already "fired" but just not called its callback yet?  It's really not clear from the CancelTimer API documentation that this is possible failure mode.  As a follow-up, please update the docs there?  And we need to audit all our CancelTimer uses.  :(

Reply:
Documented